chainreg/chainregistry.go

```diff
@@ -608,6 +608,22 @@ func NewPartialChainControl(cfg *Config) (*PartialChainControl, func(), error) {
 				return nil, nil, err
 			}
 		}
+
+	case "nochainbackend":
```

Comment:
I haven't thought about this with sufficient detail, but what are the tradeoffs between running all of lnd in `nochainbackend` mode vs trying to pull the signer out into its own binary? Is it a big lift to make the subserver into our primary rpc server?

Reply:
Yes, at the moment I think it is quite a big lift (and potentially a big code duplication issue) to extract the signer. I think the main issue would be that we'd need to fully implement the main `lnrpc.Lightning` service (to support calls like `GetInfo` and perhaps `BakeMacaroon`).

Reply:
Yeah I think pulling it out into a _distinct_ binary is something we want to do eventually, with the stop gap being thins restricted operating mode. 

Similarly, we should eventually also pull out the watch tower into a distinct binary.